include/dsn/cpp/auto_codes.h

```diff
@@ -110,6 +110,11 @@ namespace dsn
             _internal_code = 0;
         }
 
+        task_code(dsn_task_code_t code)
+        {
+            _internal_code = code;
```

Comment:
this is dangerous as implicit cast from integer to task-code becomes possible. Better to use explicit method instead, e.g., set(xxx)